Enable named tensors for arithmetic, clone, and tensor conversion ops

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23436 Make torch.view drop all names on named tensors
* #23316 Add names to repr for named tensors
* #23315 Rename tensor.is_named to has_named, expose has_named to python.
* **#23237 Enable named tensors for arithmetic, clone, and tensor conversion ops**

Featuring:
- sub add div
- torch.clone
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head

Differential Revision: [D16494416](https://our.internmc.facebook.com/intern/diff/D16494416)

zou3519 committed:
Enable named tensors for more TensorIterator ops

Featuring:
- sub add div
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

zou3519 committed:
Update on "Enable named tensors for more TensorIterator ops"

Enable named tensors for more TensorIterator ops

Featuring:
- sub add div
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head

nairbv commented:
I think we should be more specific in the title, i.e. `enabled named tensor for arithmetic and to() operators`.

Does to() actually use tensoriterator?

zou3519 commented:
> I think we should be more specific in the title, i.e. `enabled named tensor for arithmetic and to() operators`.
> 
> Does to() actually use tensoriterator?

Can do. `to()` does, it calls `copy_`: https://github.com/pytorch/pytorch/blob/2197bee3da33edccc1a6068957bf5ef7563f8012/aten/src/ATen/native/TensorConversions.cpp#L24

zou3519 committed:
Update on "Enable named tensors for more TensorIterator ops"

Enable named tensors for arithmetic, clone, and tensor conversion ops

Featuring:
- sub add div
- torch.clone
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head

zou3519 committed:
Update on "Enable named tensors for arithmetic, clone, and tensor conversion ops"

Enable named tensors for arithmetic, clone, and tensor conversion ops

Featuring:
- sub add div
- torch.clone
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head

zou3519 committed:
Update on "Enable named tensors for arithmetic, clone, and tensor conversion ops"

Enable named tensors for arithmetic, clone, and tensor conversion ops

Featuring:
- sub add div
- torch.clone
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head

zou3519 committed:
Update on "Enable named tensors for arithmetic, clone, and tensor conversion ops"

Enable named tensors for arithmetic, clone, and tensor conversion ops

Featuring:
- sub add div
- torch.clone
- tensor conversion functions that are implemented via `.to`

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 23237 gh/zou3519/77/head
